### PR TITLE
[hotfix] RescalingITCase

### PR DESCRIPTION
**Pull request by StefanRRichter**
Hotfix for RescalingITCase: race condition could make the test stuck in a blocking call until timeout.

### CODE REVIEW

File: flink-tests/src/test/java/org/apache/flink/test/checkpointing/RescalingITCase.java [hunk 1]
```diff
@@ -670,12 +676,15 @@ public void run(SourceContext<Integer> ctx) throws Exception {
 				}
 
 				Thread.sleep(100);
```
Comment (StephanEwen) [inline flink-tests/src/test/java/org/apache/flink/test/checkpointing/RescalingITCase.java]: This seems like a long wait time actually (100ms). How about waiting 2ms and signaling after the 10th element.

Comment (StefanRRichter) [inline flink-tests/src/test/java/org/apache/flink/test/checkpointing/RescalingITCase.java]: I just kept the the original value from @tillrohrmann to minimize unrelated changes. I assume the concrete choice of 100ms is arbitrary so I am fine with the proposed change.

Comment (tillrohrmann) [inline flink-tests/src/test/java/org/apache/flink/test/checkpointing/RescalingITCase.java]: You only have to make sure that the sources have not finished their processing when a savepoint is taken from them. But you can assure that differently by waiting on a future which is completed when the source has processed all a certain amount of elements.

**Comment by StephanEwen** [comment]
Looks good to me, merging this...
